src/core/core.scale.js

```diff
@@ -565,10 +565,10 @@ var Scale = Element.extend({
 	handleMargins: function() {
 		var me = this;
 		if (me.margins) {
-			me.paddingLeft = Math.max(me.paddingLeft - me.margins.left, 0);
-			me.paddingTop = Math.max(me.paddingTop - me.margins.top, 0);
-			me.paddingRight = Math.max(me.paddingRight - me.margins.right, 0);
-			me.paddingBottom = Math.max(me.paddingBottom - me.margins.bottom, 0);
+			me.paddingLeft = Math.max(me.paddingLeft, me.margins.left);
+			me.paddingTop = Math.max(me.paddingTop, me.margins.top);
+			me.paddingRight = Math.max(me.paddingRight, me.margins.right);
+			me.paddingBottom = Math.max(me.paddingBottom, me.margins.bottom);
```

Comment:
When `me.paddingLeft > me.margins.left`, the result looks incorrect as shown below:
<img width="284" alt="Screen Shot 2019-06-03 at 5 19 07 PM" src="https://user-images.githubusercontent.com/723188/58790865-c5155e00-8623-11e9-8efb-560378423229.png">

Reverting this change makes it better, but there is still a cutoff:
<img width="304" alt="Screen Shot 2019-06-03 at 5 22 41 PM" src="https://user-images.githubusercontent.com/723188/58791130-3a812e80-8624-11e9-87e5-508df34dd1a6.png">

Reply:
This was a bug in `adjustWithPadding`, fixed now. Added a config to pen for this case.